### PR TITLE
SPOI-5728: Backporting changes/improvements done by Ingestion to Malhar

### PR DESCRIPTION
**Pull request by DT-Priyanka**
This is new pull request for https://github.com/DataTorrent/Malhar/pull/1540 on ASF repository.
Replies to review comments of #1540
1. We decided to open single pull request for all consolidated requests.
2. For using concurrent collections, will create a new jira ticket to get it fixed. Application fails if we simply remote concurrent implementation right now.

### CODE REVIEW

**Comment by DT-Priyanka** [comment]
@chandnisingh for reviewer as she already reviewed #1540

**Comment by chandnisingh** [comment]
Priyanka,
Please create a separate request for each operator and assign the pull request to their authors since the person who created/worked on them is best suited to review them.

File: library/src/main/java/com/datatorrent/lib/io/block/FileReaderUtils.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  /**
+   * Converts Scheme part of the URI to lower case. Multiple URI can be comma separated. If no scheme is there, no
+   * change is made.
+   *
+   * @param
+   * @return String with scheme part as lower case
+   */
+  public static String convertSchemeToLowerCase(String uri)
+  {
```
Comment (chandnisingh) [inline library/src/main/java/com/datatorrent/lib/io/block/FileReaderUtils.java]: Aren't schemes specific to ingestion?

File: library/src/main/java/com/datatorrent/lib/io/fs/FileSplitter.java [hunk 2]
```diff
@@ -634,22 +634,23 @@ public boolean isDirectory()
...
-    private transient volatile boolean running;
+    protected transient volatile boolean running;
     protected final transient HashSet<String> ignoredFiles;
     protected transient Pattern regex;
     protected transient long sleepMillis;
 
     public TimeBasedDirectoryScanner()
     {
-      lastModifiedTimes = Maps.newHashMap();
+      lastModifiedTimes = Maps.newConcurrentMap();
```
Comment (chandnisingh) [inline library/src/main/java/com/datatorrent/lib/io/fs/FileSplitter.java]: We discussed it on the last pull request that this wasn't the right solution and ticket was also re-opened. Then why this change

**Comment by DT-Priyanka** [comment]
Closing this pull request, will open new one for each operator.
